src/application/AnimalsService.ts

```diff
@@ -17,49 +18,30 @@ export class AnimalsService {
     public async get(id: number): Promise<Animal> {
         const animal = await this.animalRepository.findOne(id);
         if (!animal) throw new ApiError('Not Found', 404, 'Animal not found in database');
-        //if (!animal) throw new Error('Animal not found in db');
+
         return animal;
     }
 
-    public async create({
-        name,
-        age,
-        specie,
-        description,
-        ready_for_adoption,
-        additional_info,
-    }: AnimalCreationParams): Promise<void> {
-        const animal = this.animalRepository.create({
-            name,
-            age,
-            specie,
-            description,
-            ready_for_adoption,
-        });
-        const {
-            temporary_home,
-            need_donations,
-            virtual_adoption,
-            adoption_date,
-            admission_to_shelter,
-            accepts_kids,
-            accepts_other_animals,
-        } = additional_info;
-        const animalAdditionalInfo = this.animalAdditionalInfo.create({
-            temporary_home,
-            need_donations,
-            virtual_adoption,
-            adoption_date,
-            admission_to_shelter,
-            accepts_kids,
-            accepts_other_animals,
-        });
+    public async create({ additionalInfo, ...animalParams }: AnimalCreationParams): Promise<void> {
+        const animal = this.animalRepository.create(animalParams);
+        const animalAdditionalInfo = this.animalAdditionalInfo.create(additionalInfo);
         animal.additional_info = animalAdditionalInfo;
 
         await this.animalRepository.save(animal);
     }
 
-    public async update(animal: Animal): Promise<Animal> {
-        throw new Error(`Not implemented ${animal}`);
+    public async update(id: number, { additionalInfo, ...animalParams }: AnimalUpdateParams): Promise<Animal> {
+        const animal = await this.animalRepository.findOne(id, { relations: ['additional_info'] });
+        if (!animal) throw new ApiError('Not Found', 404, `Animal with id: ${id} not found!`);
+        if (id % 1 !== 0) throw new ApiError('Bad Request', 400, 'Id cannot be floating point number!');
+        if (arePropertiesUndefined({ additionalInfo, ...animalParams }))
+            throw new ApiError('Bad Request', 400, 'No data provided!');
```

Comment:
Czy na pewno to jest błąd? W końcu dajemy możliwość partial update.

Reply:
Źle nazwałem funkcje, zwracam błąd jeśli ktoś poda pusty obiekt (wszystkie właściwości będą `undefined`)